lib/spack/spack/test/cmd/env.py

```diff
@@ -721,56 +740,84 @@ def test_env_with_included_config_file():
                for x in e._get_environment_specs())
 
 
-def test_env_with_included_config_scope():
+def test_env_with_included_config_file_url(tmpdir, mutable_empty_config,
+                                           packages_file):
+    """Test configuration inclusion of a file whose path is a URL before
+       the environment is concretized."""
+
+    spack_yaml = tmpdir.join('spack.yaml')
+    with spack_yaml.open('w') as f:
+        f.write(
+            "spack:\n  include:\n    - file://{0}\n".format(packages_file))
+
+    env = ev.Environment(tmpdir.strpath)
+    ev.activate(env)
+    scopes = env.included_config_scopes()
+    assert len(scopes) == 1
+
+    cfg = spack.config.get('packages')
+    assert cfg['mpileaks']['version'] == [2.2]
+
+
+def test_env_with_included_config_missing_file(tmpdir, mutable_empty_config):
+    """Test inclusion of a missing configuration file raises FetchError
+       noting missing file."""
+
+    spack_yaml = tmpdir.join('spack.yaml')
+    missing_file = tmpdir.join('packages.yaml')
+    with spack_yaml.open('w') as f:
+        f.write(
+            "spack:\n  include:\n    - {0}\n".format(missing_file.strpath))
+
+    env = ev.Environment(tmpdir.strpath)
+    with pytest.raises(FetchError, match='No such file or directory'):
+        ev.activate(env)
+
+
+def test_env_with_included_config_scope(tmpdir, packages_file):
+    """Test inclusion of a package configuration file added to an existing
+       environment's configuration directory."""
```

Comment:
(question) what is an "environment's configuration directory"? To my knowledge environments have their own config (which can exist in `spack.yaml`), or they can `include:` locations, and they use the standard configuration locations (that exist with or without environments, like `etc/spack/defaults/`), but environments do not by default maintain or search a directory that is specific to them (for example, if you place a directory called "config" in an environment's root directory, Spack won't use it by default).

If that all syncs with what you understand, then I think rewording this would be useful (in fact I think you can just e.g. replace "configuration directory" in the above docstring with "configuration"). There's a few other spots that use similar language (e.g. "standard environment configuration directory")  that would also need to be adjusted (e.g. comments in this test and also in `test_env_with_included_config_var_path`).

Reply:
The point is that remote configuration files get staged to a special subdirectory under the environment much as we do with views.  That staged directory is what I meant by the environment's configuration directory.

Hopefully my rephrasing addresses your concern here.